File: Lab2/src/Model/CompStmt.java
package Model;

public class CompStmt implements IStmt {
    private IStmt first;
    private IStmt snd;

    public CompStmt(IStmt first,IStmt snd){
        this.first = first;
        this.snd = snd;
    }


    public String toString() {
        return " (" + first.toString() + "; " + snd.toString() + ") ";
    }

    public PrgState execute(PrgState state) {
        MyIStack<IStmt> stk = state.getStk();
        stk.push(snd);
        stk.push(first);
        return state;
    }
}

File: A7/ToyLanguage/Model/CompStmt.cs
using System;
namespace ToyLanguage.Model
{
    public class CompStmt : IStmt
    {
        private IStmt first;
        private IStmt snd;

        public CompStmt(IStmt first, IStmt snd)
        {
            this.first = first;
            this.snd = snd;
        }

        public PrgState execute(PrgState state)
        {
            IStack<IStmt> stk = state.getStack();
            stk.push(snd);
            stk.push(first);
            return null;
        }
        public override string ToString()
        {
            return "(" + first.ToString() + "; " + snd.ToString() + ") ";
        }

    }
}
